rust-runtime/aws-smithy-xml/src/decode.rs

```diff
@@ -16,38 +16,64 @@ pub type Depth = usize;
 // much value in lots of different match variants
 
 #[derive(Debug)]
-pub enum XmlError {
+enum XmlDecodeErrorKind {
```

Comment:
After refactoring all these errors, I think I want to revise the RFC to state that only **actionable errors** must go into an error module. In cases where it's clear the error won't ever need to be actionable, the separate error module no longer makes sense since there is no namespace to clutter with error context structs. I'll post another PR proposing this and link it here.

Reply:
Revised in #1977.